src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/NumericConvertInstruction.cs

```diff
@@ -110,6 +105,7 @@ private object ConvertInt32(int obj)
                         case TypeCode.UInt64: return (UInt64)obj;
                         case TypeCode.Single: return (Single)obj;
                         case TypeCode.Double: return (Double)obj;
+                        case TypeCode.Decimal: return (Decimal)obj;
```

Comment:
Strictly speaking, we only need these in the `ConvertChecked` family of instructions, which are targeted by the `LightCompiler` when emitting a `Coalesce` expression. However, we could use these cases if we change the code for emitting unary `Convert` and `ConvertChecked` nodes to detect implicit conversions to `System.Decimal` from any of these source types, and emit the built-in (early-bound) numeric conversion instruction instead of the convert instruction that's parameterized on a `MethodInfo` (which evaluates slower because it makes a late-bound call to the method via reflection). Such a change is not in this PR but could be considered separately.